tools/relay-cache-warmer/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/gomodule/redigo/redis"
+)
+
+// set at build time
+var commit string
+var version string
+
+func main() {
+	var graphql string
+	var interval uint64
+	var redisURL string
+	var redisPassword string
+	var versionFlag bool
+	flag.StringVar(&graphql, "graphql", "https://graphql.grid.tf/graphql", "graphql url")
+	flag.Uint64Var(&interval, "interval", 10, "cache warming interval")
+	flag.StringVar(&redisURL, "redis-url", "redis://localhost:6379", "redis url")
+	flag.StringVar(&redisPassword, "redis-password", "", "redis password")
+	flag.BoolVar(&versionFlag, "version", false, "print version and exit")
+
+	flag.Parse()
+	if versionFlag {
+		fmt.Println(commit)
+		fmt.Println(version)
+		os.Exit(0)
+	}
+
+	var opts []redis.DialOption
+	if redisPassword != "" {
+		opts = append(opts, redis.DialPassword(redisPassword))
+	}
+
+	conn, err := redis.DialURL(redisURL, opts...)
+	if err != nil {
+		log.Fatalf("failed to connect to redis server: %s", err.Error())
+	}
```

Comment:
Never use a single connection for a long running process because u can lose this connection at any moment in time. (what if redis was restarted) 

The right way to do this is by always using a Pool and ask the pool for a connection when it's needed then close it immediately (the connection)

the pool then can decide to keep the connection actually open and use it for something else or create a new one if this one was broken, etc...